src/bitcoin-cli.cpp

```diff
@@ -480,6 +489,61 @@ static UniValue ConnectAndCallRPC(BaseRequestHandler* rh, const std::string& str
     return response;
 }
 
+/**
+ * ReplaceAtHeightByBlockHash
+ *
+ * @param method  Name of the method to be called
+ * @param args    Vector of arguments to be modified
+ */
+static void ReplaceAtHeightByBlockHash(const std::string& method, std::vector<std::string>& args, bool named)
+{
+    // avoid any recursion risk
+    if (method == "getblockhash" || method == "getbestblockhash") return;
+
+    for (size_t i = 0; i < args.size(); ++i) {
+        std::string name;
+        Span<const char> value = args[i];
+
+        if (named) {
+            size_t pos = args[i].find('=');
+            if (pos == std::string::npos) continue;
+            auto namespan = value.subspan(0, pos);
+            name = std::string(namespan.data(), namespan.size());
+            if (!RPCConvertNamedBlockhash(method, name)) continue;
+            value = value.subspan(pos + 1);
+        } else {
+            if (!RPCConvertBlockhash(method, i)) continue;
+        }
+        if (value.size() <= 1 || value[0] != '@') continue;
+
+        std::string bh_method;
+        std::vector<std::string> bh_args;
+        if (args[i] != "@best") {
+            bh_method = "getbestblockhash";
+        } else {
+            bh_method = "getblockhash";
+            std::string bh_arg(value.data()+1, value.size()-1);
+            if (named) {
+                bh_arg = "height=" + bh_arg;
+            }
+            bh_args.push_back(bh_arg);
+        }
+
+        DefaultRequestHandler rh;
+        const UniValue reply = ConnectAndCallRPC(&rh, bh_method, bh_args);
+        const UniValue& result = find_value(reply, "result");
+        const UniValue& error = find_value(reply, "error");
+        if (error.isNull() && result.isStr()) {
+            if (named) {
+                args[i] = name + "=" + result.get_str();
+                fprintf(stderr, "set args[i] = %s\n", args[i].c_str());
```

Comment:
It appears this line is getting caught in the linter:
```
The locale dependent function fprintf(...) appears to be used:
src/bitcoin-cli.cpp:                fprintf(stderr, "set args[i] = %s\n", args[i].c_str());

Unnecessary locale dependence can cause bugs that are very
tricky to isolate and fix. Please avoid using locale dependent
functions if possible.

Advice not applicable in this specific case? Add an exception
by updating the ignore list in ./test/lint/lint-locale-dependence.sh
```

Reply:
Good linter, that line shouldn't have made it past debugging.